### PR TITLE
changed google provider to latest version 

### PR DESCRIPTION
**Pull request by emiioan**
changed google provider to latest version and also implemented make apply to Makefile

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,6 +26,10 @@ lint: check_shell check_shebangs check_python check_golang check_terraform \
 create:
 	@source $(ROOT)/scripts/create.sh
 
+.PHONY: apply
```
Comment (cristi-) [inline Makefile]: Could you add a note in the [troubleshooting](https://github.com/GoogleCloudPlatform/gke-rbac-demo#troubleshooting) section about when to use this target (e.g. timeout on cluster creation/resume on failures)?

**Comment by emiioan** [comment]
@googlebot I singed it!

**Comment by emiioan** [comment]
@googlebot I fixed it.

**Comment by emiioan** [comment]
@googlebot I fixed it.

**Comment by chrislovecnm** [comment]
@emiioan you need to change the git email address in the commit, please.

**Pull request closed without merging**